tests/group/test_group_result_parser.py

```diff
@@ -62,6 +64,37 @@ def test_group_admin_throws_error_on_add_existing_group(
             + f"'{group_name}' already exists as a '{self.group_admin._profile_type}' profile.",
         )
 
+    def test_group_admin_add_surfaces_error_from_preliminary_extract(
+        self,
+        call_racf_mock: Mock,
+    ):
+        group_name = "TESTGRP0"
+        extract_group_error_xml = (
+            TestGroupConstants.TEST_EXTRACT_GROUP_RESULT_BASE_ONLY_ERROR_XML
+        )
+        extract_group_error_xml = extract_group_error_xml.replace(
+            "<message>ICH51003I NAME NOT FOUND IN RACF DATA SET</message>",
+            "<message>NOT A REAL ERROR MESSAGE</message>",
+        )
+        call_racf_mock.side_effect = [
+            extract_group_error_xml,
+            TestGroupConstants.TEST_ADD_GROUP_RESULT_SUCCESS_XML,
+        ]
+        with self.assertRaises(SecurityRequestError) as exception:
+            self.group_admin.add(
+                group_name, traits=TestGroupConstants.TEST_ADD_GROUP_REQUEST_TRAITS
+            )
+        extract_group_error_json = copy.deepcopy(
+            TestGroupConstants.TEST_EXTRACT_GROUP_RESULT_BASE_ONLY_ERROR_DICTIONARY
```

Comment:
I don't think you need `copy.deepcopy` here. I am pretty sure that you can just use `dict(my_dictionary)` to create a deep copy.

Reply:
Unfortunately, this did not work. I did try that first. The issue is because the dictionary contains other dictionaries `dict(my_dictionary)` was making a shallow copy, which meant that the modification made to `extract_group_error_json` was carried to other values in TEST_EXTRACT_GROUP_RESULT_BASE_ONLY_ERROR_DICTIONARY.